wifi-subsys/components/tools/src/serialization.c

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2022 Mesh4all <mesh4all.org>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @{
+ *
+ * @file
+ * @brief       Cbor functions to encode and decode values
+ *
+ * @author      Locha Mesh Developers (contact@locha.io)
+ *
+ */
+#include  <stdio.h>
+#include "serialization.h"
+#include "esp_log.h"
+
+static const char* TAG = "SERIALIZATION";
+uint8_t buf[100];
+
+esp_err_t encode(uint8_t* coded_msg, CborValue *iter)
+{
+    CborEncoder root_encoder;
+    CborParser root_parser;
+    CborError err = CborNoError;
+
+    // Initialize the outermost cbor encoder
+    cbor_encoder_init(&root_encoder, buf, sizeof(buf), 0);
+    err = cbor_encode_text_stringz(&root_encoder, (char *)coded_msg);
+    ESP_LOGI(TAG, "encoded buffer size %d", cbor_encoder_get_buffer_size(&root_encoder, buf));
+    if (err != CborNoError) {
+        ESP_LOGE(TAG, "error encoding: %d ", err);
+        return ESP_FAIL;
+    }
+    // Initialize the cbor parser and the value iterator
+    err = cbor_parser_init(buf, sizeof(buf), 0, &root_parser, iter);
+    if (err != CborNoError) {
+        ESP_LOGE(TAG, "error paser_init encoding: %d ", err);
+        return ESP_FAIL;
+    }
+    err = cbor_value_to_json(stdout, iter, 0);
+    if (err != CborNoError) {
+        ESP_LOGE(TAG, "error converting in json:  %d ", err);
+        return ESP_FAIL;
+    }
+    return ESP_OK;
+}
+
+esp_err_t decode(CborValue *iter, char* decoded_msg)
```

Comment:
for a successfully  decode you have to pass exactly length to decode, other stuff is what happens if is need pass a parameter more long that 100 bytes? please do more dinamyc the function.